### PR TITLE
refactor: refactoring `hideSidebar` configuration.

### PR DESCRIPTION
**Pull request by Koooooo-7**
Refactor `hideSidebar` that removing the `tpl` of sidebar (if `hideSidebar : true`).
Checking the `hideSidebar` when having options on `sidebar`.

**Summary**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [x] Docs
- [ ] Build-related changes
- [ ] Repo settings
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

### CODE REVIEW

**Review by anikethsaha — APPROVED** [review]
This is a much better approach than the current one. 

cc @sy-records , this will fix https://github.com/docsifyjs/docsify/issues/1392 as well.

**Review by anikethsaha — CHANGES_REQUESTED** [review]
Can you write tests for this if possible ?

**Comment by Koooooo-7** [comment]
> Can you write tests for this if possible ?

yup, I will check the test tutorial and try to get a new test. 😅

**Review by Koooooo-7 — COMMENTED** [review]
Add test and we may add more tests of configurations here.
The CI issues cuz the test site includes `search plugin` to cause the error.

File: src/plugins/search/search.js [hunk 1]
```diff
@@ -226,6 +226,9 @@ export function init(config, vm) {
         namespaceSuffix = matches[0];
       }
     }
+    paths.unshift(namespaceSuffix + '/');
+  } else {
+    paths.unshift('/');
```
Comment (sy-records) [inline src/plugins/search/search.js:231]: Merge error, your code shouldn't contain my code.

**Review by sy-records — CHANGES_REQUESTED** [review]

**Review by anikethsaha — APPROVED** [review]

**Comment by sy-records** [comment]
```
    TypeError: Cannot read property 'insertBefore' of null

      163 | 
      164 |   function before(target, el) {
    > 165 |     return target.insertBefore(el, target.children[0]);
          |                   ^
      166 |   }
      167 | 
      168 |   function on(el, type, handler) {
```
Error due to [docsify-plugin-carbon](https://github.com/waruqi/docsify-plugin-carbon) plugin, @waruqi Can you fix it in your plugin?

**Comment by Koooooo-7** [comment]
@sy-records  I think this is not the issue of carbon plugin , cuz when u hide the sidebar, how can u use this carbon plugin or other plugins which r based on the sidebar.

**Comment by sy-records** [comment]
Plugins are not compatible with the hidden sidebar

**Comment by Koooooo-7** [comment]
> Plugins are not compatible with the hidden sidebar

so, do u wanna hide the sidebar and use those plugins which r based on the sidebar? it makes no sense.

**Comment by sy-records** [comment]
So I just wanted to make the plugin side compatible.

**Comment by waruqi** [comment]
> ```
>     TypeError: Cannot read property 'insertBefore' of null
> 
>       163 | 
>       164 |   function before(target, el) {
>     > 165 |     return target.insertBefore(el, target.children[0]);
>           |                   ^
>       166 |   }
>       167 | 
>       168 |   function on(el, type, handler) {
> ```
> 
> Error due to [docsify-plugin-carbon](https://github.com/waruqi/docsify-plugin-carbon) plugin, @waruqi Can you fix it in your plugin?

Ok, I will look at it.

**Comment by waruqi** [comment]
> > ```
> >     TypeError: Cannot read property 'insertBefore' of null
> > 
> >       163 | 
> >       164 |   function before(target, el) {
> >     > 165 |     return target.insertBefore(el, target.children[0]);
> >           |                   ^
> >       166 |   }
> >       167 | 
> >       168 |   function on(el, type, handler) {
> > ```
> > 
> > 
> > Error due to [docsify-plugin-carbon](https://github.com/waruqi/docsify-plugin-carbon) plugin, @waruqi Can you fix it in your plugin?
> 
> Ok, I will look at it.

@sy-records I have fixed it, please update to docsify-plugin-carbon@1.0.2

```
<script src="//cdn.jsdelivr.net/npm/docsify-plugin-carbon@1.0.2/index.min.js"></script>
```

File: src/plugins/search/search.js [hunk 2]
```diff
@@ -226,9 +226,6 @@ export function init(config, vm) {
         namespaceSuffix = matches[0];
       }
     }
-    paths.unshift(namespaceSuffix + '/');
```
Comment (sy-records) [inline src/plugins/search/search.js:229]: Merge error...

File: test/e2e/sidebar.test.js [hunk 3]
```diff
@@ -1,16 +0,0 @@
...
-  // ---------------------------------------------------------------------------
-  test('search readme', async () => {
-    await docsifyInit();
-    await page.goto(DOCS_URL + '/#/quickstart');
-    await page.fill('input[type=search]', 'Please consider donating');
-    expect(
-      await page.innerText('.results-panel > .matching-post > a > h2')
-    ).toEqual('Donate');
-  });
-});
```
Comment (sy-records) [inline test/e2e/sidebar.test.js:16]: Merge error

**Review by sy-records — CHANGES_REQUESTED** [review]

**Comment by Koooooo-7** [comment]
@waruqi  thx ur input ! IMHO,  literally, it is not ur duty to make the plugin which is based on sidebar to suit the no sidebar case, it seems awkward.
@sy-records As I mentioned, it is not the carbon plugin caused some err, could u provide any information about it ?

**Comment by Koooooo-7** [comment]
cc @anikethsaha  @sy-records  I fixed the CI issue, review it again plz.

**Review by anikethsaha — APPROVED** [review]

File: src/core/render/index.js [hunk 4]
```diff
@@ -128,18 +128,7 @@ export function renderMixin(proto) {
...
-    const { maxLevel, subMaxLevel, loadSidebar, hideSidebar } = this.config;
-
-    if (hideSidebar) {
-      // FIXME : better styling solution
-      document.querySelector('aside.sidebar').remove();
-      document.querySelector('button.sidebar-toggle').remove();
-      document.querySelector('section.content').style.right = 'unset';
-      document.querySelector('section.content').style.left = 'unset';
-      document.querySelector('section.content').style.position = 'relative';
-      document.querySelector('section.content').style.width = '100%';
```
Comment (sy-records) [inline src/core/render/index.js:140]: @anikethsaha Should we keep the style?

### develop branch

image

### now 

image

**Review by sy-records — CHANGES_REQUESTED** [review]

**Comment by sy-records** [comment]
Please check the console for some errors

image

**Comment by Koooooo-7** [comment]
> Please check the console for some errors
> 
> image

Plz Checking the exception again and make sure how it is reproducible and what it is blame for.FYI. If you just found something wrong/exception without catching the point,  it would be useless.

**Comment by sy-records** [comment]
Add `hideSidebar: true,` to index.html to reproduce.

**Comment by Koooooo-7** [comment]
@sy-records sorry, I can't make it reproducible, could u please do me a favor that checking about it.

**Comment by sy-records** [comment]
Well, ignore it.

**Comment by Koooooo-7** [comment]
> Well, ignore it.

what does the `ignore` mean?  if there were an issue, we should figure it out to make our site changes and test reliable.

File: src/core/render/index.js (hunk 4 above)
Comment (Koooooo-7) [reply src/core/render/index.js:140]: I think we can just keep this configurations without adjusting much styles right now （dont let render code including much styles）, and if there has much users proposal to make it that we can refine it again.
WDYT?

Comment (sy-records) [reply src/core/render/index.js:140]: I think we should keep them and not change them in the z versions.

**Comment by jhildenbiddle** [comment]
@Koooooo-7 --

Is this PR still worth considering or should we close it?

**Comment by Koooooo-7** [comment]
> @Koooooo-7 --
> 
> Is this PR still worth considering or should we close it? I admit to not reviewing it due to the age, but I'm happy to if you'd like me to.

It seems outdate, I'd like to close it for now,

**Pull request closed without merging**